src/govuk/components/modal-dialogue/modal-dialogue.js

```diff
@@ -0,0 +1,186 @@
+import { nodeListForEach } from '../../common'
+import '../../vendor/polyfills/Element/prototype/classList'
+import '../../vendor/polyfills/Function/prototype/bind'
+import '../../vendor/polyfills/Event' // addEventListener and event.target normaliziation
+
+function ModalDialogue ($module) {
+  this.$module = $module
+  this.$dialogBox = $module.querySelector('dialog')
+  this.$container = document.documentElement
+
+  // Check for browser support
+  this.hasNativeDialog = 'showModal' in this.$dialogBox
+
+  // Allowed focussable elements
+  this.focussable = [
+    'button',
+    '[href]',
+    'input',
+    'select',
+    'textarea'
+  ]
+}
+
+// Initialize component
+ModalDialogue.prototype.init = function (options) {
+  this.options = options || {}
+
+  this.open = this.handleOpen.bind(this)
```

Comment:
I wonder if these method behinds should be in the constructor instead?

Reply:
@nickcolley I try to defer anything that uses more CPU cycles until it's actually needed (like when `.init()` gets run) but it's probably negligible so don't mind.

Reply:
Thanks Colin 👍 We believe that the performance impact is negligible so we should make the change for consistency with other components.